gatsby-browser.js

```diff
@@ -1,7 +1,15 @@
+import * as React from "react"
+import {StoreProvider} from "./src/components/ga4/EnhancedEcommerce/store-context"
 import CustomLayout from "./gatsby/wrapRootElement.js"
+import "./src/styles/ecommerce/variables.css"
 
 // TODO - look into making this work like gatsby-node & use typescript for the
 // things that are imported/exported.
 
 export { onInitialClientRender } from "./gatsby/onInitialClientRender"
 export const wrapPageElement = CustomLayout
+
+// Wrap every page using a StoreProvider object used by eCommerce demo.
+export const wrapRootElement = ({ element }) => (
+    <StoreProvider>{element}</StoreProvider>
```

Comment:
Is this store provider actually needed for all pages? My gut is that it'd be better to put this provider at the future src/pages/ecommerce page instead.

Reply:
Although I agree, I am unable to find another way to get the same instance of StoreProvider shared between different pages (e.g. product-index and product-details). If I move  StoreProvider from the wrapper to, say, Layout component, each page gets its own instance of StoreProvider which obviously results in the cart being lost once you navigate between pages. I'll try to find a way to limit she scope of StoreProvider to the ecommerce demo only, but have not figured this out yet.

Reply:
Ah, I see. 

If you have actual distinct pages (in ./src/pages/) this would be the only way to do this via useContext. That's _probably_ fine, but another option that might be worth exploring is to manage that state in redux, which is already wired in via [here](https://github.com/googleanalytics/ga-dev-tools/blob/main/gatsby/wrapRootElement.js#L15). This isn't a silver bullet either, so I'm happy with whatever decision you come to on the tradeoffs of keeping it in different places.

Reply:
Oh one last thing to consider. Since this is already a firebase project, it should be fairly trivial to start using firestore if you want to persist cart data.